liberty-maven-plugin/src/main/resources/META-INF/plexus/components.xml

```diff
@@ -20,17 +20,47 @@
                     <lifecycle>
                         <id>default</id>
                         <phases>
+                            <pre-clean>
+                                net.wasdev.wlp.maven.plugins:liberty-maven-plugin:stop-server
+                            </pre-clean>
                             <process-resources>
-                                org.apache.maven.plugins:maven-resources-plugin:resources,
-                                net.wasdev.wlp.maven.plugins:liberty-maven-plugin:create-server
+                                org.apache.maven.plugins:maven-resources-plugin:3.0.2:resources
                             </process-resources>
                             <compile>
-                                net.wasdev.wlp.maven.plugins:liberty-maven-plugin:install-apps,
-                                net.wasdev.wlp.maven.plugins:liberty-maven-plugin:install-feature
+                                org.apache.maven.plugins:maven-compiler-plugin:3.6.1:compile
                             </compile>
-                            <package>net.wasdev.wlp.maven.plugins:liberty-maven-plugin:package-server</package>
-                            <install>org.apache.maven.plugins:maven-install-plugin:install</install>
-                            <deploy>org.apache.maven.plugins:maven-deploy-plugin:deploy</deploy>
+                            <process-test-resources>
+                                org.apache.maven.plugins:maven-resources-plugin:3.0.2:testResources
+                            </process-test-resources>
+                            <test-compile>
+                                org.apache.maven.plugins:maven-compiler-plugin:3.6.1:testCompile
+                            </test-compile>
+                            <test>
+                                org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test
+                            </test>
+                            <prepare-package>
+                                net.wasdev.wlp.maven.plugins:liberty-maven-plugin:create-server,
+                                net.wasdev.wlp.maven.plugins:liberty-maven-plugin:install-feature,
+                                net.wasdev.wlp.maven.plugins:liberty-maven-plugin:install-apps
+                            </prepare-package>
+                            <package>
+                                net.wasdev.wlp.maven.plugins:liberty-maven-plugin:package-server
+                            </package>
+                            <pre-integration-test>
+                                net.wasdev.wlp.maven.plugins:liberty-maven-plugin:start-server
```

Comment:
Yeah, good point.  At one time I was thinking that some logic could be added to not start the server under some conditions such as no tests available.   I will verify that the pom can choose to skip the start/stop.